src/test/test_pidfile.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -e
+
+#
+# test pidfile here 
+#
+
+# Includes
+source ../qa/workunits/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:17108"
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    CEPH_ARGS+="--mon-host=$CEPH_MON "
+
+    local funcs=${@:-$(set | sed -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        $func $dir || return 1
+    done
+}
+
+
+function TEST_normal() {
```

Comment:
this test may not be the most perfect solution, but own its uniqe functionity.

//thanks trociny , i take your reply as a "lgtm". i agree that a test for pidfile.cc would be ideal because we can use SubProcess and a little daemon binary for testing purpose. probably also because of the fact that it's hard to cleanup a certain process without knowing its pid: that's what teardown_unexist_pidfile() is trying to do. it could kill innocent processes living in the same system.